### PR TITLE
Leverage module system to type check and document `makeOutputs` arguments

### PR DESCRIPTION
**Pull request by antotocar34**
Defines a module in `./src/modules/flake-parts/makeOutputsArgs.nix` , which defines an option for each argument that the `makeOutputs` function expects.

I've left TODO comments around the place where I'm not certain about things. 

Goes some way towards fixing #320

### CODE REVIEW

File: src/modules/flake-parts/makeOutputsArgs.nix [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    };
+
+    # TODO(antotocar34) make an enum of all available subsystems?
+    subsystem = mkOption {
+      description = ''name of subsystem to use. Examples: rust, python, nodejs'';
+      example = "nodejs";
+      type = t.str;
+    };
+
+    subsystemInfo = mkOption {
```
[the remark this answers is not in the document]
Comment (antotocar34) [reply src/modules/flake-parts/makeOutputsArgs.nix:32]: If they are indeed just translator arguments I don't think `subsystemInfo` is a very insightful name. The word `info` to mean doesn't really indicate that something needs to be passed in to do something.

File: src/modules/flake-parts/makeOutputsArgs.nix [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      description = "the projects found by the discoverer";
+    };
+
+    pname = mkOption {
+      default = null;
+      type = t.nullOr t.str;
+      description = "package name";
+    };
+
+    settings = mkOption {
```
[the remark this answers is not in the document]
Comment (antotocar34) [reply src/modules/flake-parts/makeOutputsArgs.nix:57]: Added a note in the description to reflect this.

**Comment by antotocar34** [comment]
Test fails because of 
https://github.com/nix-community/dream2nix/blob/ce047a9ceaf1040dc14e393cc3243d22dd0dbcdc/examples/python_poetry/flake.nix#L27-L37

error msg:
```
error: The option `perSystem.x86_64-linux.dream2nix.inputs.my-project.projects.my-project.pythonVersion' does not exist. Definition values:
       - In `<unknown-file>, via option perSystem': "3.10"
(use '--show-trace' to show detailed location information)
testing example for python_poetry
```

File: src/modules/flake-parts/makeOutputsArgs.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (DavHau) [reply src/modules/flake-parts/makeOutputsArgs.nix:32]: Agree. By pulling all arguments up into the top level we will get rid of `subsystemInfo` completely. ... problem solved ;)

File: src/modules/flake-parts/makeOutputsArgs.nix [hunk 3]
```diff
@@ -70,45 +67,61 @@ in {
...
-        ];
-        "@tiptap/extension-code"."2.0.0-beta.26" = [
-          ["@tiptap/core" "2.0.0-beta.174"]
-        ];
-      };
-    };
+  inject = mkOption {
+    default = {};
+    type = t.lazyAttrsOf (t.listOf (t.listOf t.str));
+    description = "Inject missing dependencies into the dependency tree"; # TODO(antotocar34) find a suitable description
```
Comment (DavHau) [inline src/modules/flake-parts/makeOutputsArgs.nix]: TODO can be removed

**Comment by DavHau** [comment]
Thanks a lot!

**Pull request merged**